### PR TITLE
48 baseline model training lstm

### LINKED ISSUE #48: BaseLine Model Training
### **Feature: Baseline Model Training Implementation**

**Description:**
Implement training scripts for baseline models for each data type. Each script should handle model training, evaluation, metrics logging, and visualizations according to project standards.

---

**Acceptance Criteria / Requirements:**

1. **Train Script Location & Naming:**

   * Reside in `project_root/train/`
   * Naming convention: `baseline_train_<model_used>.py`

2. **Output & Storage:**

   * Results should be stored in:
     `project_root/experiments/baseline/baseline_results_<model_used>/`
   * Save model evaluation metrics as **JSON**:

     * MAE
     * MSE
     * RMSE
     * MAPE
     * SMAPE

3. **Training Logs & Visuals:**

   * Save **epoch vs train/test loss** as PNG
   * Save **checkpoints** in the same folder
   * Save **hyperparameters** used as JSON, e.g.:

     ```json
     {"n_epochs": 10, "hidden_size": 4, ...}
     ```
   * Save **model summary**
   * Generate **visualization plots** (up to 10) showing:

     * Date vs stock price
     * Input price, predicted price, actual price

4. **Model-Specific Considerations:**

   * For time series models: standardize input values (strategy can be custom)
   * Implement **early stopping** with clear criteria

---

**Notes:**

* All outputs must follow the folder structure for reproducibility.
* Plots and metrics should be clearly labeled for ease of analysis.
* Ensure scripts are modular and easy to extend to other baseline models.

### PR DESCRIPTION
**Pull request by pranavdhawann**
### Results Summary

| Metric | **Model1** | **Model2** | **Model3** |
|--------|-------------------------|----------------------------------------------|-----------------------------------|
| **MSE** | 41.27 | **37.54** | 42.74 |
| **RMSE** | $6.42 | **$6.13** | $6.54 |
| **MAE** | $3.18 | **$3.06** | $3.32 |
| **MAPE** | **6.69%** | 7.20% | 8.04% |
| **sMAPE** | **5.11%** | 5.40% | 5.95% |
| **Best Epoch** | 43 / 58 | 60 / 75 | 44 / 59 |
| **Features** | 5 | 16 | 16 |
| **Parameters** | 275K | 277K | 294K |

### CODE REVIEW

File: reports/experimental-results/lstm/combined/combined_config.json
Comment (aash1999) [inline reports/experimental-results/lstm/combined/combined_config.json]: Can you move reports/experimental-results/lstm to experiments/baseline/lstm
